### PR TITLE
fix #14875 - fix shp filename sanitization and update test

### LINKED ISSUE #14875: Tests fail with dBase extension enabled
### Describe the bug

PHPUnit tests fail with dBase PHP extension enabled.

Affects [QA_4_9](https://github.com/phpmyadmin/phpmyadmin/tree/QA_4_9) and [master](https://github.com/phpmyadmin/phpmyadmin/tree/master) branches.

### To Reproduce

1. Install [dBase](https://pecl.php.net/package/dbase) extension
2. Run PHPUnit

### Additional context

https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/481601035#L821

```
1) PhpMyAdmin\Tests\Plugins\Import\ImportShpTest::testImportOsm
Failed asserting that PhpMyAdmin\Message Object &000000005b19305f000000003ebfb4da (
    'number' => 8
    'string' => 'There was an error importing the ESRI shape file: "%s".'
    'message' => ''
    'isDisplayed' => false
    'useBBCode' => true
    'hash' => null
    'params' => Array &0 (
        0 => 'It wasn't possible to find the DBase file &quot;/home/travis/build/phpmyadmin/phpmyadmin/tmp//shp/dresden_osm_shp_gis_osm_transport_a_v06.dbf&quot;'
    )
    'addedMessages' => Array &1 ()
) matches expected ''.

/home/travis/build/phpmyadmin/phpmyadmin/test/classes/Plugins/Import/ImportShpTest.php:84
/home/travis/build/phpmyadmin/phpmyadmin/test/classes/Plugins/Import/ImportShpTest.php:143

2) PhpMyAdmin\Tests\Plugins\Import\ImportShpTest::testDoImport
Failed asserting that PhpMyAdmin\Message Object &000000005b19345a000000003ebfb4da (
    'number' => 8
    'string' => 'There was an error importing the ESRI shape file: "%s".'
    'message' => ''
    'isDisplayed' => false
    'useBBCode' => true
    'hash' => null
    'params' => Array &0 (
        0 => 'It wasn't possible to find the DBase file &quot;/home/travis/build/phpmyadmin/phpmyadmin/tmp//shp/bc_hospitals.dbf&quot;'
    )
    'addedMessages' => Array &1 ()
) matches expected ''.

/home/travis/build/phpmyadmin/phpmyadmin/test/classes/Plugins/Import/ImportShpTest.php:84
/home/travis/build/phpmyadmin/phpmyadmin/test/classes/Plugins/Import/ImportShpTest.php:172
```

### PR DESCRIPTION
**Pull request by bepsvpt**
### Description

Origin sanitization will remove the filename extension and it will result in file not found when loading the content([referece](https://travis-ci.org/phpmyadmin/phpmyadmin/jobs/633026890#L569-L581)).

This pull request fixes shp filename sanitization bug. 

Fixes #14875

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [x] Any new functionality is covered by tests.

### CODE REVIEW

File: libraries/classes/Plugins/Import/ImportShp.php [hunk 1]
```diff
@@ -114,21 +114,18 @@ public function doImport(array &$sql_data = [])
...
                     );
                     if ($extracted !== false) {
-                        $dbf_file_path = $temp . (PMA_IS_WINDOWS ? '\\' : '/')
-                            . Sanitize::sanitizeFilename($dbf_file_name, true);
-                        $handle = fopen($dbf_file_path, 'wb');
-                        if ($handle !== false) {
-                            fwrite($handle, $extracted);
-                            fclose($handle);
+                        // remove filename extension
+                        $dbf_file_name = substr($dbf_file_name, 0, -4);
```
Comment (williamdes) [inline libraries/classes/Plugins/Import/ImportShp.php]: I would suggest using https://www.php.net/manual/en/function.pathinfo.php

**Review by williamdes — CHANGES_REQUESTED** [review]
Thank you for the fix !
I added a last comment to improve the code

File: libraries/classes/Plugins/Import/ImportShp.php [hunk 2]
```diff
@@ -223,7 +221,7 @@ public function doImport(array &$sql_data = [])
 
                 if ($shp->getDBFHeader() !== null) {
                     foreach ($shp->getDBFHeader() as $c) {
-                        $cell = trim($record->DBFData[$c[0]]);
+                        $cell = trim(strval($record->DBFData[$c[0]]));
```
Comment (williamdes) [inline libraries/classes/Plugins/Import/ImportShp.php]:
```suggestion
                        $cell = trim((string) $record->DBFData[$c[0]]);
```

**Review by williamdes — APPROVED** [review]

**Comment by williamdes** [comment]
@bepsvpt Do you want to signoff or that I use your changes using my signoff line ?

**Comment by bepsvpt** [comment]
Hi @williamdes,

You can use your signoff line.

I'm curious about if the repository owner(organization) can not verify the contributer's signed-off. Is there any different between using real name or nickname(fake name)?

**Comment by williamdes** [comment]
Hi @bepsvpt 

Thank you for your contributions !

I forwarded your question to the team

**Comment by bepsvpt** [comment]
Thanks a lot. :)

**Pull request closed without merging**